### PR TITLE
rustup: update to 1.23.1.

### PR DESCRIPTION
**Pull request by jcgruenhage**

### CODE REVIEW

**Comment by jcgruenhage** [comment]
So, rustup now fails *earlier* than before for armv6-musl. Before, it failed on post_build, now it already fails on do_build. Should we just mark rustup as broken on armv6-musl?

**Comment by ericonr** [comment]
Yeah, broken is probably correct, I think...

That or `archs="~armv6l-musl"`

File: srcpkgs/rustup/template [hunk 1]
```diff
@@ -12,7 +13,7 @@ maintainer="Jan Christian Grünhage <jan.christian@gruenhage.xyz>"
 license="Apache-2.0, MIT"
 homepage="https://www.rustup.rs"
 distfiles="https://github.com/rust-lang/${pkgname}/archive/${version}.tar.gz"
-checksum=ad46cc624f318a9493aa62fc9612a450564fe20ba93c689e0ad856bff3c64c5b
+checksum=0203231bfe405ee1c7d5e7e1c013b9b8a2dc87fbd8474e77f500331d8c26365f
 
 post_build() {
 	# rustup errors out because it doesn't know about armv6l-musl
```
Comment (ericonr) [inline srcpkgs/rustup/template:19]: If it won't be built for armv6l-musl anymre, the special casing around it below should be fixed as well

File: srcpkgs/rustup/template [hunk 2]
```diff
@@ -1,7 +1,8 @@
 # Template file for 'rustup'
 pkgname=rustup
-version=1.22.1
-revision=2
+version=1.23.1
+revision=1
+archs="~armv6l-musl"
```
Comment (ericonr) [inline srcpkgs/rustup/template:5]:
```suggestion
# rustup doesn't recognize this as a target
archs="~armv6l-musl"
```

**Comment by Piraty** [comment]
#27759

**Comment by Piraty** [comment]
ab6576648f121037ddbcb6c5fd19064e7abe2554 armv7-musl is broken as well which went unnoticed as CI doesn't build armv7-musl

**Pull request merged**